### ISSUE TITLE
enableDangerousExperimentalLeanBuilds needs to be marked as unsupported for SDK 41+ projects

### ISSUE THREAD

**Issue by AbdnhVmpr** [expo/expo#12583]
## 🐛 Bug Report

**package.json**

```
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "expo": "^41.0.0",
    "expo-status-bar": "~1.0.4",
    "expo-notifications": "^0.11.5",
    "expo-splash-screen": "^0.10.2",
    "expo-updates": "^0.5.4",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-41.0.0.tar.gz",
    "react-native-web": "~0.13.12"
  },
  "devDependencies": {
    "@babel/core": "~7.9.0"
  },
  "private": true
}
```

**app.json**

```
{
  "expo": {
    "name": "my_app",
    "slug": "my_app",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": ["**/*"],
    "android": {
      "package": "com.myapp",
      "versionCode": 1,
      "permissions": [""],
      "enableDangerousExperimentalLeanBuilds": true
    }
  }
}
```

The app works fine in development, but when I add **enableDangerousExperimentalLeanBuilds** in app.json file and I build the App Bundle, the application crashes after installing and opening it on my device. 

I have already had the exact same issue before #10569. But I could fix it by installing these package: 
**expo-notifications expo-updates expo-splash-screen**

Now I'm still having the same problem even tough those packages have been already installed

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

41

### Environment

```
Expo CLI 4.4.1 environment info:
    System:
      OS: Windows 10 10.0.17763
    Binaries:
      Node: 12.14.1 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  4.1.0.0 AI-201.8743.12.41.7042882
    npmPackages:
      expo: ^41.0.0 => 41.0.0
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-41.0.0.tar.gz => 0.63.2
      react-native-web: ~0.13.12 => 0.13.18
    Expo Workflow: managed
```

### Reproducible demo or steps to reproduce from a blank project

The problem doesn't happen while development.
It happens when I build the app and install it on my device.
Larger apps see lower install and update success rates and take up device storage on user's devices, that's why I'm using **enableDangerousExperimentalLeanBuilds**

**Comment by brentvatne** [comment]
hi there!

i don't recommend using this field with sdk 41. we are going to flag it as unsupported in sdk 41+ next week, we are releasing managed support on eas build which works much better than this flag ever did, and so it's no longer needed.

**Comment by AbdnhVmpr** [comment]
Is there any temporary solution for this issue? 
Because I have to update my apps in play store as soon as possible. At the moment I won't be able to wait for a week in order to use the managed support on eas build

**Comment by brentvatne** [comment]
the solution is to remove the `enableDangerousExperimentalLeanBuilds` - this was never something that was stable and reliable ("dangerous" and "experimental" in the name 😄 ). as for eas build -- you don't have to wait - you can try it on eas build now and it should work well. there aren't a lot of changes planned between now and when we announce the preview. if you encounter any issues, let us know.

**Comment by AbdnhVmpr** [comment]
If I remove the `enableDangerousExperimentalLeanBuilds ` the app bundle size becomes more than 55 MB which is too big, the users won't download the app just because of it's size.

The EAS build is only available in priority plan. That means I won't be able to reduce the app size as long as I'm using the free plan?

**Comment by brentvatne** [comment]
your options are:

- use EAS Build. it has one month free, then small monthly fee. within a few months we'll roll out the free tier for it, and you can use that if that works for you. if you can't afford the monthly fee, let us know and we'd be happy to give you free access until the free trial rolls out, in exchange for your feedback from using the service!
- roll back with SDK 40 to use `enableDangerousExperimentalLeanBuilds`.
- attempt to debug why the build isn't working with `enableDangerousExperimentalLeanBuilds` on SDK 41.
- eject your project and build it on your own machine -- this is very similar to what EAS Build does for you.

**Comment by AbdnhVmpr** [comment]
Thank you for your response.

1. Sorry but I won't be able to use EAS Build

2. I have been using SDK 40, but lately I got a notification from google play saying that:
they have detected that my apps contain the `requestLegacyExternalStorage` flag in the manifest file, developers with apps on devices running Android 11+ must use scoped storage to give users better access control over their device store. So before May 5th I must update my app to declare the All files access `MANAGE_EXTERNAL_STORAGE`. That's why I want to update all my apps to the latest SDK 41.

3. Can you please help me to how to debug why `enableDangerousExperimentalLeanBuilds` isn't working on SDK 41 ?

4. Ejecting the app isn't the best option for me :( fortunately I'm not able to do it

**Comment by brentvatne** [comment]
> I have been using SDK 40, but lately I got a notification from google play saying that:
they have detected that my apps contain the requestLegacyExternalStorage flag in the manifest file, developers with apps on devices running Android 11+ must use scoped storage to give users better access control over their device store. So before May 5th I must update my app to declare the All files access MANAGE_EXTERNAL_STORAGE. That's why I want to update all my apps to the latest SDK 41.

this is a vague error message that doesn't require action on your behalf: https://forums.expo.io/t/starting-may-5th-you-must-let-us-know-why-your-app-requires-broad-storage-access/51156/9?u=notbrent

> hi there! it’s hard to parse out exactly what google wants from us here. you can read more discussion on this in this reddit thread: https://www.reddit.com/r/androiddev/comments/mqzls8/ominous_scoped_storage_warning_messages/ if you are have an expo managed workflow app, as far as i know there is no action needed on your behalf. i imagine almost every app on the play store would have received this same warning email. we’ll investigate further and let people know what we find
>
> we’ll follow up with a more detailed blog post soon, but it does seem like no action is needed on your part

sdk 40 targets api 29, not 30, and so you can continue without changing anything. there is more discussion on this in the reddit thread, but a response from google reads:

>  Kindly note that this temporary uploading restriction only affects apps that both target Android 11 (API level 30) and request the all files access permission.

sdk 40 does not target api 30, so this does not apply to you.

**Comment by brentvatne** [comment]
> Sorry but I won't be able to use EAS Build

i don't understand

> Can you please help me to how to debug why enableDangerousExperimentalLeanBuilds isn't working on SDK 41 ?

if you can share a link to your project build page i can take a quick look but i can't promise anything and i would not depend on this in the future.

**Comment by AbdnhVmpr** [comment]
Thank you for your effort, I really appreciate it.
My username on expo is: thevampirex

Where should I give my feedback as an exchange from using the service?

**Comment by brentvatne** [comment]
@AbdnhVmpr - that account doesn't appear to have any builds associated with it, are you sure you want to add it there?

**Comment by AbdnhVmpr** [comment]
Yes. I've just created this account. I want to start building new apps in it

**Comment by brentvatne** [comment]
just so you know, there is no way to "transfer" projects between accounts at the moment, so this won't solve the issue for your existing app.

**Comment by AbdnhVmpr** [comment]
For this existing app I will rollback to SDK 40 as you've mentioned and I'll continue without changing anything

**Comment by brentvatne** [comment]
sounds good! we'll add access to your account when we're back in the office on monday.

**Comment by AbdnhVmpr** [comment]
Thank you Brent for your time, I really appreciate it

**Comment by brentvatne** [comment]
no worries @AbdnhVmpr - have a great weekend!

**Comment by Ghassanee** [comment]
> 
> 
> no worries @AbdnhVmpr - have a great weekend!

Can you add my account too @aka_ghassane 
Thanks ! ☺
